app/chart/load-config.js

```diff
@@ -513,27 +513,256 @@ var configs = {
+                        logger.log({
+                            level: 'error',
+                            message: 'error while fetching config',
+                            errMsg: JSON.parse(body).message,
+                        });
+                        reject(JSON.parse(body).message);
+                    }
+
+                });
+            } catch (error) {
+                logger.log({
+                    level: 'error',
+                    message: 'error while fetching config',
+                    errMsg: error.message,
+                });
+                reject(error);
+            }
+
+        });
+
+        return chartconfigPromise;
+
+    },
+    KPIChartConfig: function (viz_id) {
+
+        var chartconfigPromise = new Promise((resolve, reject) => {
+
+            try {
+                request(vizMetaApi + "/" + viz_id, function (error, response, body) {
+                    if (error) {
+                        reject(error.message);
+                        return;
+                    }
+                    if (response && response.statusCode == 200) {
+                        var json_res = JSON.parse(body);
+                        var properties = json_res.visualMetadata.properties;
+                        var fields = json_res.visualMetadata.fields;
+                        var result = {};
+                        var features = VisualizationUtils.getDimensionsAndMeasures(fields),
+                            dimension = features.dimensions,
+                            measures = features.measures;
+                        result['dimension'] = VisualizationUtils.getNames(dimension);
+                        result['measure'] = VisualizationUtils.getNames(measures);
+                        result['kpiDisplayName'] = [];
+                        result['kpiAlignment'] = [];
+                        result['kpiBackgroundColor'] = [];
+                        result['kpiNumberFormat'] = [];
+                        result['kpiFontStyle'] = [];
+                        result['kpiFontWeight'] = [];
+                        result['kpiFontSize'] = [];
+                        result['kpiColor'] = [];
+                        result['kpiColorExpression'] = [];
+                        result['kpiIcon'] = [];
+                        result['kpiIconFontWeight'] = [];
+                        result['kpiIconColor'] = [];
+                        result['kpiIconExpression'] = [];
+                        result['FontSizeforDisplayName'] = [];
+                        for (var i = 0; i < measures.length; i++) {
+                            result['kpiDisplayName'].push(
+                                VisualizationUtils.getFieldPropertyValue(measures[i], 'Display name') ||
+                                result['measure'][i]
+                            );
+                            result['kpiAlignment'].push(VisualizationUtils.getFieldPropertyValue(measures[i], 'Text alignment'));
```

Comment:
props "Text alignment" are coming from API so can we use that instead of hard coding them

Reply:
I have used the same format to extract chart info, which is used in all chart-service in flair-bi.